### PR TITLE
[New validate] GR108

### PR DESCRIPTION
**Pull request by shmuel44**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11528

## Description
Moved GR108 validation to the new validate format. The validation ensures that non-hidden packs do not have mandatory dependencies on hidden packs.

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/GR_validators/GR108_is_invalid_packs_dependencies_valid.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    ValidationResult,
+)
+
+ContentTypes = Pack
+
+
+class IsInvalidPacksDependenciesValidator(BaseValidator[ContentTypes], ABC):
+    error_code = "GR108"
+    description = "Validates that non-hidden packs do not have mandatory dependencies on hidden packs."
+    rationale = "Hidden packs should not be critical dependencies to ensure proper pack management."
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/GR_validators/GR108_is_invalid_packs_dependencies_valid.py:19]:
```suggestion
    rationale = "Hidden packs are not available to install in the marketplace."
```

File: demisto_sdk/commands/validate/validators/GR_validators/GR108_is_invalid_packs_dependencies_valid.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    error_message = "Pack {dependent_pack} has mandatory dependencies on hidden pack(s): {hidden_packs}"
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/GR_validators/GR108_is_invalid_packs_dependencies_valid.py:20]:
```suggestion
    error_message = "Pack {dependent_pack} has hidden pack(s) {hidden_packs} in its mandatory dependencies"
```

**Review by dorschw — COMMENTED** [review]
nice. add a jira, fix the small comments and we'll be ready to merge

File: demisto_sdk/commands/validate/tests/GR_validators_test.py [hunk 3]
```diff
@@ -529,3 +535,100 @@ def test_IsUsingUnknownContentValidator__different_dependency_type__list_files(
...
+    graph_interface = repo_for_test_gr_108.create_graph()
+    BaseValidator.graph_interface = graph_interface
+    results = (
+        IsInvalidPacksDependenciesValidatorListFiles().obtain_invalid_content_items(
+            [repo_for_test_gr_108.packs[0]]
+        )
+    )
+    assert (
+        results[0].message
+        == "Pack Pack1 has mandatory dependencies on hidden pack(s): Pack2"
```
Comment (shmuel44) [inline demisto_sdk/commands/validate/tests/GR_validators_test.py:626]:
```suggestion
        == "Pack Pack1 has hidden pack(s) Pack2 in its mandatory dependencies"
```

File: demisto_sdk/commands/validate/tests/GR_validators_test.py [hunk 4]
```diff
@@ -529,3 +535,100 @@ def test_IsUsingUnknownContentValidator__different_dependency_type__list_files(
...
+        The validator should return a result indicating that Pack1 depends on the hidden Pack2.
+    """
+    graph_interface = repo_for_test_gr_108.create_graph()
+    BaseValidator.graph_interface = graph_interface
+    results = (
+        IsInvalidPacksDependenciesValidatorAllFiles().obtain_invalid_content_items([])
+    )
+    assert (
+        results[0].message
+        == "Pack Pack1 has mandatory dependencies on hidden pack(s): Pack2"
```
Comment (shmuel44) [inline demisto_sdk/commands/validate/tests/GR_validators_test.py:603]:
```suggestion
        == "Pack Pack1 has hidden pack(s) Pack2 in its mandatory dependencies"
```

**Review by dorschw — APPROVED** [review]

**Pull request merged**